File: lib/rest/yampi/mappers/yampi_brand_mapper.dart
import 'package:sertton/core/catalog/dtos/brand_dto.dart';
import 'package:sertton/rest/types/json.dart';

class YampiBrandMapper {
  static BrandDto toDto(Json json) {
    return BrandDto(id: json['id'].toString(), name: json['name']);
  }

  static List<BrandDto> toDtoList(Json json) {
    final data = (json['data'] as List).cast<Json>();
    return data.map(toDto).toList();
  }
}

File: src/rest/yampi/mappers/yampi-brand-mapper.ts
import type { Brand } from "@/core/catalog/entities"
import type { YampiBrand } from "@/rest/yampi/types"

export const YampiBrandMapper = () => ({
  toDomain(input: YampiBrand): Brand {
    return { id: String(input.id), name: input.name }
  },
})
